packages/skeleton-svelte/src/lib/components/Tab/TabsControl.svelte

```diff
@@ -0,0 +1,126 @@
+		// Tab
+		tabBase = '',
+		tabRounded = 'rounded-container',
+		tabOutline = 'focus:outline focus:outline-primary-600-400 focus:outline-offset-4',
+		tabClasses = '',
+		// Tab Content
+		contentBase = 'flex',
+		contentSpaceX = 'space-x-1',
+		contentClasses = '',
+		// Events
+		onclick = getContext<(event: MouseEvent) => {}>('onclick'),
+		onkeypress = getContext<(event: KeyboardEvent) => {}>('onkeypress'),
+		onkeydown = getContext<(event: KeyboardEvent) => {}>('onkeydown'),
+		onkeyup = getContext<(event: KeyboardEvent) => {}>('onkeyup'),
+		onchange = getContext<(group: string) => {}>('onchange'),
+		// Snippets
+		children
+	}: TabsControlProps = $props();
+
+	const selected = $derived(group === name);
+	const rxActive = $derived(selected ? active : inactive);
+
+	let elemInput: HTMLInputElement;
+
+	function onTabKeyDown(event: KeyboardEvent) {
+		onkeydown(event);
+		if (!['Enter', 'Space', 'ArrowRight', 'ArrowLeft', 'Home', 'End'].includes(event.code)) return;
+
+		event.preventDefault();
+
+		if (['Enter', 'Space'].includes(event.code)) {
+			elemInput.click();
+			return;
+		}
+
+		const currTab = elemInput.closest('[role="tab"]');
+		if (!currTab) return;
+
+		const tabList = elemInput.closest('[role="tablist"]');
+		if (!tabList) return;
+
+		const isRTL = getComputedStyle(tabList).direction === 'rtl';
+		const tabs = Array.from(tabList.querySelectorAll('[role="tab"]'));
+		const currIndex = tabs.indexOf(currTab);
+
+		let nextIndex = -1;
+		switch (event.code) {
+			case 'ArrowRight':
+				if (isRTL) {
+					nextIndex = currIndex - 1 < 0 ? tabs.length - 1 : currIndex - 1;
+				} else {
+					nextIndex = currIndex + 1 >= tabs.length ? 0 : currIndex + 1;
+				}
+				break;
+			case 'ArrowLeft':
+				if (isRTL) {
+					nextIndex = currIndex + 1 >= tabs.length ? 0 : currIndex + 1;
+				} else {
+					nextIndex = currIndex - 1 < 0 ? tabs.length - 1 : currIndex - 1;
+				}
+				break;
+			case 'Home':
+				nextIndex = 0;
+				break;
+			case 'End':
+				nextIndex = tabs.length - 1;
+				break;
+		}
+
+		if (nextIndex < 0) return;
+
+		const nextTab = tabs![nextIndex!];
+		const nextTabInput = nextTab?.querySelector('input');
+		if (nextTabInput) {
+			nextTabInput.click();
+			(nextTab as HTMLDivElement).focus();
+		}
+	}
+</script>
+
+<!-- @component A Tab Control component. -->
+
+<label class="{base} {text} {justify} {rxActive} {background} {padding} {rounded} {spacingY} {cursor} {classes}" {title}>
+	<li
+		aria-controls={controls}
+		tabindex={selected ? 0 : -1}
+		aria-selected={selected}
+		data-testid="tabs-control"
+		role="tab"
+		class="{tabBase} {tabRounded} {tabOutline} {tabClasses}"
+		onkeydown={onTabKeyDown}
+		{onkeypress}
+		{onkeyup}
+	>
+		<div class="h-0 w-0 flex-none overflow-hidden">
+			<input bind:this={elemInput} type="radio" bind:group {name} value={name} onchange={() => onchange(group)} {onclick} tabindex="-1" />
+		</div>
+		{#if children}
+			<div class="{contentBase} {contentSpaceX} {contentClasses}">
```

Comment:
This extra div seems redundant. We already have so many layers here. It's like an onion heh.

Reply:
this helps providing styles for the children (flex, grid, etc..), I think it is nice to have and doesn't require much maintenance, otherwise when users wants to have tabs with text and icons they will have to wrap it every time.
let me know if this doesn't convince you and I will delete it :+1: